File: Source/Samples/Jacobi.Vst.Samples.Delay.VB/PluginPrograms.vb
Imports Jacobi.Vst.Framework
Imports Jacobi.Vst.Framework.Plugin

Namespace Jacobi.Vst.Samples.Delay.VB

    Friend Class PluginPrograms
        Inherits VstPluginProgramsBase

        Private _plugin As FxTestPlugin

        Public Sub New(ByVal plugin As FxTestPlugin)
            _plugin = plugin
        End Sub

        Protected Overrides Function CreateProgramCollection() As Framework.VstProgramCollection
            Dim programs As VstProgramCollection = New VstProgramCollection()

            Dim prog As VstProgram = New VstProgram(_plugin.ParameterFactory.Categories)
            prog.Name = "Fx Program 1"
            _plugin.ParameterFactory.CreateParameters(prog.Parameters)

            programs.Add(prog)

            prog = New VstProgram(_plugin.ParameterFactory.Categories)
            prog.Name = "Fx Program 2"
            _plugin.ParameterFactory.CreateParameters(prog.Parameters)

            programs.Add(prog)

            prog = New VstProgram(_plugin.ParameterFactory.Categories)
            prog.Name = "Fx Program 3"
            _plugin.ParameterFactory.CreateParameters(prog.Parameters)

            programs.Add(prog)

            Return programs
        End Function

    End Class

End Namespace

File: Source.Core/Samples/Jacobi.Vst.Samples.Delay/PluginPrograms.cs
namespace Jacobi.Vst.Samples.Delay
{
    using Jacobi.Vst.Framework;
    using Jacobi.Vst.Framework.Plugin;

    /// <summary>
    /// This class manages the plugin programs.
    /// </summary>
    class PluginPrograms : VstPluginProgramsBase
    {
        Plugin _plugin;

        /// <summary>
        /// Constructs a new instance.
        /// </summary>
        /// <param name="plugin">Must not be null.</param>
        public PluginPrograms(Plugin plugin)
        {
            _plugin = plugin;
        }

        /// <summary>
        /// Initializes the plugin program collection.
        /// </summary>
        /// <returns>A filled program collection.</returns>
        protected override VstProgramCollection CreateProgramCollection()
        {
            VstProgramCollection programs = new VstProgramCollection();

            VstProgram prog = new VstProgram(_plugin.ParameterFactory.Categories);
            prog.Name = "Fx Program 1";
            _plugin.ParameterFactory.CreateParameters(prog.Parameters);

            programs.Add(prog);

            prog = new VstProgram(_plugin.ParameterFactory.Categories);
            prog.Name = "Fx Program 2";
            _plugin.ParameterFactory.CreateParameters(prog.Parameters);

            programs.Add(prog);

            prog = new VstProgram(_plugin.ParameterFactory.Categories);
            prog.Name = "Fx Program 3";
            _plugin.ParameterFactory.CreateParameters(prog.Parameters);

            programs.Add(prog);

            return programs;
        }
    }
}
